pkg/cloud/services/compute/instance.go

```diff
@@ -471,14 +484,25 @@ func (s *Service) getOrCreatePort(eventObject runtime.Object, clusterName string
 		TenantID:            portOpts.TenantID,
 		ProjectID:           portOpts.ProjectID,
 		SecurityGroups:      securityGroups,
-		AllowedAddressPairs: []ports.AddressPair{},
+		AllowedAddressPairs: addressPairs,
 	}
 
-	for _, ap := range portOpts.AllowedAddressPairs {
-		createOpts.AllowedAddressPairs = append(createOpts.AllowedAddressPairs, ports.AddressPair{
-			IPAddress:  ap.IPAddress,
-			MACAddress: ap.MACAddress,
-		})
+	var finalCreateOpts ports.CreateOptsBuilder
+	finalCreateOpts = createOpts
```

Comment:
Can't we instead directly set the createOpts as ports.CreateOptsBuilder so we don't have to define the `finalCreateOpts`?

Reply:
yeah, I can clean this up a little